compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.desktop.kt

```diff
@@ -550,6 +553,8 @@ internal class ComposeSceneMediator(
     }
 
     override fun onRender(canvas: Canvas, width: Int, height: Int, nanoTime: Long) = catchExceptions {
+        interopContainer.executeScheduledUpdates()
```

Comment:
We'll have one-frame lag:

```
recompose
draw Compose

vsync
changes Swing state
draw Swing
```

I would move it to the end of the function, and reset scheduling (as we do in ComposeScene/postponeInvalidation)

Reply:
I think I'm missing the essence of what you are trying to propose. Can you elaborate?

Reply:
1. Move `executeScheduledUpdates` to the end of `onRender`, so we can perform interop changes in this frame
2. Because we perform them in this frame, we don't need to request a new frame (`requestRedraw`).
3. We only need to `requestRedraw` when the task is scheduled outside of `composeScene.render` function. It happens when:
- the state is changed
- it is placed/unplaced during recomposition on a scroll event in LazyColumn.

Something like this:
```
override fun onRender(canvas: Canvas, width: Int, height: Int, nanoTime: Long) = catchExceptions {
        willExecuteScheduled {
              canvas.withSceneOffset {
                  scene.render(asComposeCanvas(), nanoTime)
              }
        }
        interopContainer.executeScheduledUpdates()
}

private var willExecuteScheduled = false

override fun scheduleUpdate(action: () -> Unit) {
    scheduledUpdatesSwapchain.scheduleUpdate(action)

    if (!willExecuteScheduled) {
        requestRedraw()
    }
}

private fun willExecuteScheduled(action: () -> Unit) {
  willExecuteScheduled = true
  try {
    action()
  } finally {
    willExecuteScheduled = false
  }
}
```

Reply:
Aha, the idea of `requestRedraw` was we could possible update state in a way, that is not detected in Compose (because it doesn't read it) and thus no invalidation happens. In this example it is detected (and I'm not sure why).

```
    var color by remember {
        mutableStateOf(Color.Red)
    }

    Thread {
        while (true) {
            Thread.sleep(1000)
            color = Color((0..255).random(), (0..255).random(), (0..255).random())
        }
    }.start()

    SwingPanel(
        background = Color.White,
        factory = {
            JPanel()
        },
        modifier = Modifier.fillMaxSize(),
        update = {
            it.background = color.toAwtColor()
        }
    )
```

Reply:
What you propose, basically, is that some of the changes to state could be done during recomposition itself, so we need to do it first and only then to actually executed scheduled changes. Got it.

Reply:
Done in https://github.com/JetBrains/compose-multiplatform-core/pull/1489/commits/9d3e63026f32eff5caeb9eefa25fcac02a496e60